src/components/DataTypes/boolean.tsx

```diff
@@ -1,12 +1,13 @@
+import { OperatorKeys } from '.';
 import type { CreateFilter } from './utils';
 
 export const operators = () => ({
 	is: 'is',
 });
 
-export type OperatorSlug = keyof ReturnType<typeof operators>;
+export type OperatorSlugs = OperatorKeys<typeof operators>;
 
-export const createFilter: CreateFilter<OperatorSlug | 'is_not'> = (
+export const createFilter: CreateFilter<OperatorSlugs | 'is_not'> = (
```

Comment:
We don't have `is_not`since we want to have a single operator in the operator's dropdown in the filter modal when we filter by a boolean. I'm not sure if I got what you mean but `full text search` cannot be in the operators otherwise we would see it in the dropdown choices. Did I get your point or am I in the wrong direction ?

Reply:
I probably confused you. I agree on implementing only the operators that makes sense for each data type (ie I don't think contains for Boolean makes 100% sense. I meant:
```
export type OperatorSlug = keyof ReturnType<typeof operators> | 'is_not';

export const createFilter: CreateFilter<OperatorSlug> = (
```
Essentially having OperatorSlug define all the operators that the CreateFilter supports, even if the `operators` map defines just a subset (like in this case, which we do not want to surface an `is_not` operator in the dropdown (actually let's add a comment about this so it's clear to everyone why `operators` only has `is`, assuming that's indeed the reason)).

Reply:
@thgreasi unfortunately I have seen that there is an underlying problem. Having removed the operator converter from `array.tsx` (change which I honestly love :) ) I find myself having to handle every case in natural types. The case I have in mind is the following:

```
Data = {... numberArray: [1,2,3,4]}

Schema = {
...
  "numberArray": {
     "type": "arrays",
     "items": {
       "type": "number"
     }
  }
}
```

In this case there is no `refScheme`, so we would have array operators (contains, does not contain) to select one of the numbers inside the array, and this would cause an empty filter if we don't handle the contains case in the `number.tsx`